src/ar_meta_db.erl

```diff
@@ -1,91 +1,179 @@
+%% @doc Defines a small in-memory metadata table for Arweave nodes.
+%% Typically used to store small peices of globally useful information
+%% (for example: the port number used by the node).
+%% @end
+
 -module(ar_meta_db).
--export([start/0, get/1, put/2, keys/0, remove_old/1,remove_old/2, increase/2]).
+-behaviour(gen_server).
+-define(SERVER, ?MODULE).
+
 -compile({no_auto_import, [{get, 1}, {put, 2}]}).
 -include_lib("eunit/include/eunit.hrl").
 -include("ar.hrl").
-%%% Defines a small in-memory metadata table for Arweave nodes.
-%%% Typically used to store small peices of globally useful information
-%%% (for example: the port number used by the node).
-
-%% @doc Initialise the metadata storage service.
-start() ->
-	spawn(
-		fun() ->
-			ar:report([starting_meta_db]),
-			ets:new(?MODULE, [set, public, named_table, {read_concurrency, true}]),
-			ets:new(blacklist, [set, public, named_table]),
-			receive stop -> ok end
-		end
-	),
-	% Add a short wait to ensure that the table has been created
-	% before returning.
-	receive after 250 -> ok end.
-
-%% @doc Put an Erlang term into the meta DB. Typically these are
-%% write-once values.
-put(Key, Val) -> ets:insert(?MODULE, {Key, Val}).
-%% @doc Retreive a term from the meta db.
+
+-export([get/1, put/2, keys/0, remove_old/1, increase/2]).
+-export([start_link/0, stop/0]).
+-export([init/1,
+	 handle_call/3,
+	 handle_cast/2,
+	 handle_info/2,
+	 terminate/2,
+	 code_change/3]).
+
+%%------------------------------------------------------------------------------
+%% API
+%%------------------------------------------------------------------------------
+
+%% @doc Insert key-value-pair into db
+put(Key, Value) ->
+    gen_server:call(?MODULE, {put, Key, Value}).
+
+%% @doc Retreive value for key
 get(Key) ->
-	case ets:lookup(?MODULE, Key) of
-		[{Key, Obj}] -> Obj;
-		[] -> not_found
-	end.
+    gen_server:call(?MODULE, {get, Key}).
```

Comment:
We use tabs for indentation.

Reply:
I hopefully changed it now. If it is wrong please let me know.

Reply:
Thank you.  Only `case`s still seem to have some extra indents.

Reply:
Is it better now? Sorry for the miss.

Reply:
A consistent formatting would look like this:
```
...
        Res = case ets:lookup(?MODULE, Key) of
                [{Key, Obj}] -> Obj;
	        [] -> not_found
        end,
        {reply, Res, State};
```

Reply:
I am sorry for this. I need to setup my emacs to work with tabs instead of spaces.

Reply:
Are you sure that the end should be on the same level as Res? According to my "style" layout, it should be on the same level as the case. However, I am unsure what is correct.

Reply:
I do not have a strong preference. We have [a styleguide](https://github.com/ArweaveTeam/arweave/blob/master/arweave_styleguide.md) but it does not define how to handle this. In other places we align the way I posted, if I remember correctly.

Reply:
I think it has something to do that I assign Res using a case. To be honest, I don't what is best. I just follow emacs :joy:

Reply:
I found an example in LearnYouSomeErlang book.

```
%% note, this one would be better as a pattern match in function heads!
%% I'm doing it this way for the sake of the example.
help_me(Animal) ->
    Talk = if Animal == cat  -> "meow";
              Animal == beef -> "mooo";
              Animal == dog  -> "bark";
              Animal == tree -> "bark";
              true -> "fgdadfgna"
           end,
    {Animal, "says " ++ Talk ++ "!"}.
```